lms/djangoapps/certificates/models.py

```diff
@@ -1264,3 +1264,43 @@ class Meta:
 
     def __str__(self):
         return str(self.arguments)
+
+@python_2_unicode_compatible
+class CertificateDateOverride(TimeStampedModel):
+    """
+    Model to manually override a given certificate date with the given date.
+
+    .. no_pii:
+    """
+    generated_certificate = models.OneToOneField(
```

Comment:
A note on `OneToOneField`s: 

"A **`RelatedObjectDoesNotExist`** exception is raised when accessing the reverse relationship if an entry in the related table doesn’t exist. This is a subclass of the target model’s **`Model.DoesNotExist`** exception. For example, if a [certificate] doesn’t have [an override date] designated by [**`CertificateDateOverride`**]:

```
    # doc example edited for our purposes
    **`>>>** certificate.date_override
    Traceback (most recent call last):
        ...
    RelatedObjectDoesNotExist: Certificate has no date_override.`
```

That means `certificate.date_override` would raise this exception *in most cases*. We’ll need to handle for this with `try`/ `except` blocks, or use `hasattr`. Is this enough of a pain to go to a `ForeignKey` relationship? Examples:

```
    try:
    	certificate.date_override
    except RelatedObjectDoesNotExist:
    	print("There is no date override for this certificate")
    	# do something else

    # or

    hasattr(certificate, 'date_override')
    > False
```

- Docs: [https://docs.djangoproject.com/en/3.1/ref/models/fields/#django.db.models.OneToOneField](https://docs.djangoproject.com/en/3.1/ref/models/fields/#django.db.models.OneToOneField)

Reply:
We would usually use the try/except here.

Reply:
From discussion at Standup: we don’t mind this. Stick with the `OneToOneField`, handle the exceptions. ✅ (If this becomes really annoying, we can add an instance method that does the check for us.)